core/src/main/java/com/sequenceiq/cloudbreak/service/publicendpoint/FreeIPAEndpointManagementService.java

```diff
@@ -0,0 +1,117 @@
+                    request.setCname(endpoint);
+                    request.setDnsZone(domain);
+                    request.setTargetFqdn(StringUtils.appendIfMissing(loadBalancer.getDns(), DOMAIN_PART_DELIMITER));
+                    request.setEnvironmentCrn(stack.getEnvironmentCrn());
+                    dnsV1Endpoint.addDnsCnameRecord(request);
+                } else if (StringUtils.isNotEmpty(loadBalancer.getIp())) {
+                    AddDnsARecordRequest request = new AddDnsARecordRequest();
+                    request.setHostname(endpoint);
+                    request.setDnsZone(domain);
+                    request.setIp(loadBalancer.getIp());
+                    request.setEnvironmentCrn(stack.getEnvironmentCrn());
+                    dnsV1Endpoint.addDnsARecord(request);
+                } else {
+                    LOGGER.error("Unable to find DNS or IP for load balancer. Load balancer will not be registered with FreeIPA.");
+                }
+            }
+        }
+    }
+
+    public void deleteLoadBalancerDomainFromFreeIPA(Stack stack) {
+        if (manageCertificateAndDnsInPem()) {
+            Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stack.getId());
+            if (!loadBalancers.isEmpty()) {
+                Optional<LoadBalancer> loadBalancerOptional = selectLoadBalancer(loadBalancers);
+                if (loadBalancerOptional.isEmpty()) {
+                    LOGGER.debug("Unabled to determine load balancer type. Load balancer will not be deleted from FreeIPA.");
+                    return;
+                }
+
+                LoadBalancer loadBalancer = loadBalancerOptional.get();
+                String endpoint = loadBalancer.getEndpoint();
+                String domain = getDomainFromStack(stack);
+
+                if (StringUtils.isNotEmpty(loadBalancer.getDns())) {
+                    dnsV1Endpoint.deleteDnsCnameRecord(stack.getEnvironmentCrn(), domain, endpoint);
+                } else if (StringUtils.isNotEmpty(loadBalancer.getIp())) {
+                    dnsV1Endpoint.deleteDnsARecord(stack.getEnvironmentCrn(), domain, endpoint);
+                } else {
+                    LOGGER.debug("Unable to find DNS or IP for load balancer. Load balancer will not be deleted from FreeIPA.");
+                }
+            }
+        }
+    }
+
+    private String getDomainFromStack(Stack stack) {
+        String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
+        DetailedEnvironmentResponse environment = environmentClientService.getByCrn(stack.getEnvironmentCrn());
+        return getDomainNameProvider().getDomainName(environment.getName(), getWorkloadSubdomain(userCrn));
+    }
+
+    private Optional<LoadBalancer> selectLoadBalancer(Set<LoadBalancer> loadBalancers) {
+        Optional<LoadBalancer> loadBalancerOptional = loadBalancers.stream()
+            .filter(lb -> lb.getType() == LoadBalancerType.PRIVATE)
```

Comment:
we are using here lb.getType() == LoadBalancerType.PRIVATE other places using equals in your PR. I think it would be nice to use the same mechanism.